### PR TITLE
Removed meteo extractor

### PR DESCRIPTION
**Pull request by GriffinBabe**
@HansVRP I removed the METEO fetcher as the same functionalities are present in the METEO fetcher

### CODE REVIEW

**Review by VincentVerelst — COMMENTED** [review]
One question: this does mean we don't support fetching meteo from the openeo STAC API anymore. Should we then maybe add an assertion that if "AGERA5" is selected, only the terrascope backend can be used?

**Comment by GriffinBabe** [comment]
@VincentVerelst I added the functionality `build_generic_extractor_stac` to continue supporting stac extraction, and added some checks based on the backend.

From a discussion with @HansVRP we decided that GFMAP shouldn't take too much care of checking if a collection is valid or not in a certain backend but rather instead let the `load_collection` method raise the exception itself.

**Comment by GriffinBabe** [comment]
@VincentVerelst can you review this PR ?

File: src/openeo_gfmap/fetching/generic.py [hunk 1]
```diff
@@ -28,18 +28,36 @@
...
 ) -> Callable:
     band_mapping: Optional[dict] = None
 
     if collection_name == "COPERNICUS_30":
         band_mapping = BASE_DEM_MAPPING
     elif collection_name == "AGERA5":
         band_mapping = BASE_WEATHER_MAPPING
+    elif is_stac and (AGERA5_TERRASCOPE_STAC in collection_name):
+        if backend not in [Backend.TERRASCOPE, Backend.FED]:
+            raise ValueError(
```
Comment (VincentVerelst) [inline src/openeo_gfmap/fetching/generic.py:57]: Should this error not be thrown if `is_stac==False` and the backend is not Terrascope or Fed?

Comment (GriffinBabe) [reply src/openeo_gfmap/fetching/generic.py]: Yes you are right, I removed the check. Actually:

> From a discussion with @HansVRP we decided that GFMAP shouldn't take too much care of checking if a collection is valid or not in a certain backend but rather instead let the load_collection method raise the exception itself.

**Review by VincentVerelst — APPROVED** [review]

**Pull request merged**